tests/bindings/test_statevector_nb.py

```diff
@@ -0,0 +1,364 @@
+
+        num_qubits = 1
+        # Create a numpy array representing |0> state with appropriate size
+        dtype = np.complex128 if precision == "128" else np.complex64
+        state_data = np.zeros(2**num_qubits, dtype=dtype)
+        state_data[0] = 1.0
+
+        # Create state vector with just number of qubits
+        sv = StateVectorClass(num_qubits)
+        # Then update data
+        sv.updateData(state_data)
+
+        # Get the result
+        result = np.zeros(2**num_qubits, dtype=dtype)
+        sv.getState(result)
+        print("Result before matrix application: ", result)
+
+        # Apply a matrix to the state vector
+        matrix = np.array([[0, 1], [1, 0]], dtype=dtype)  # X gate
+        sv.applyMatrix(matrix, [0], False)
+
+        # Get the result
+        result = np.zeros(2**num_qubits, dtype=dtype)
+        sv.getState(result)
+
+        # Expected: |1⟩ state
+        expected = np.zeros(2**num_qubits, dtype=dtype)
+        expected[1] = 1.0
+
+        # Assert the result matches the expected state
+        np.testing.assert_allclose(result, expected)
+
+    def test_statevector_update_data(self, precision, nanobind_module, get_statevector_class):
+        """Test updateData method on StateVectorC64/128 classes for lightning.qubit_nb."""
+        module = nanobind_module("qubit")
+
+        StateVectorClass = get_statevector_class(module, precision)
+
+        num_qubits = 2
+        # Create a numpy array with custom data
+        dtype = np.complex128 if precision == "128" else np.complex64
+        state_data = np.zeros(2**num_qubits, dtype=dtype)
+        state_data[1] = 1.0  # |01⟩ state
+
+        # Create state vector with just number of qubits
+        sv = StateVectorClass(num_qubits)
+        # Then update data
+        sv.updateData(state_data)
+
+        # Get the result
+        result = np.zeros(2**num_qubits, dtype=dtype)
+        sv.getState(result)
+
+        # Assert the result matches the input data
+        np.testing.assert_allclose(result, state_data)
+
+    @pytest.mark.parametrize("backend_name", backends)
+    def test_statevector_reset(
+        self, backend_name, precision, nanobind_module, get_statevector_class
+    ):
+        """Test resetStateVector method on StateVectorC64/128 classes."""
+        module = nanobind_module(backend_name)
+
+        StateVectorClass = get_statevector_class(module, precision)
+
+        num_qubits = 2
+        # Create a numpy array with non-zero state
+        dtype = np.complex128 if precision == "128" else np.complex64
+        state_data = np.zeros(2**num_qubits, dtype=dtype)
+        state_data[1] = 1.0  # |01⟩ state
+
+        # Initialize with number of qubits first
+        sv = StateVectorClass(num_qubits)
+        # Then update data
+        sv.updateData(state_data)
+
+        # Get the result
+        result = np.zeros(2**num_qubits, dtype=dtype)
+        sv.getState(result)
+
```

Comment:
```suggestion
        # Get the result
        result = np.zeros(2**num_qubits, dtype=dtype)
        sv.getState(result)
        assert np.allclose(result, state_data)
    
```

Reply:
I prefer comparing with the expected data, as below.